### PR TITLE
test: set arm outputs to az devops variable group (🚧)

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds integration tests for the PS script that sets the ARM outputs as variables in Azure DevOps variable groups.

Relates to #174

### CODE REVIEW

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.DevOps.tests.ps1 [hunk 1]
```diff
@@ -20,7 +20,7 @@ InModuleScope Arcus.Scripting.DevOps {
                 try {
                     # Act
                     Save-AzDevOpsBuild -ProjectId $projectId -BuildId $buildId
-
+
```
Comment (fgheysels) [inline src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.DevOps.tests.ps1:23]:
```suggestion

```
What nitpickers we are ... :)

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.DevOps.tests.ps1 [hunk 2]
```diff
@@ -33,5 +33,40 @@ InModuleScope Arcus.Scripting.DevOps {
...
+                    # Act
+                    Set-AzDevOpsArmOutputsToVariableGroup -VariableGroupName $existingVariableGroup
+
+                    # Assert
+                    $getVariableGroupUrl= $projectUri + $project + "/_apis/distributedtask/variablegroups?api-version=" + $apiVersion + "&groupName=" + $existingVariableGroup
+                    $variableGroup = Invoke-RestMethod -Uri $getVariableGroupUrl -Headers $headers -Verbose
+                    $variableGroup = $variableGroup.value[0]
+                    $variableGroup.variables.$variableName | Should -Not -Be $null
+                    $variableGroup.variables.$variableName.value | Should -Be "@{value=$variableValue}"
+                #} finally {
```
Comment (fgheysels) [inline src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.DevOps.tests.ps1:61]: Why is the try/finally construct commented out ?

**Review by fgheysels — APPROVED** [review]
Looks OK to me.

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.DevOps.tests.ps1 (hunk 2 above)
Comment (stijnmoreels) [reply src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.DevOps.tests.ps1:61]: Because I'm still working on it. This PR is not ready for a review.

**Comment by stijnmoreels** [comment]
I have a problem with giving access rights to the build user. Normally only two users are to have write permissions on the Azure DevOps variable group, but that doesn't seem to be enough. Could it be that the request comes from a GitHub repo instead of a Azure DevOps repo? Maybe some additional user requires permissions, only not sure who that would be.

Currently, the tests fail with `Forbidden`.

**Comment by pim-simons** [comment]
Ran into the same issue while create an integration test in this PR https://github.com/arcus-azure/arcus.scripting/pull/252.
While updating the variable group the test receives back an HTTP 403.

It turns out that when you start the pipeline manually it does work, so there seems to be an issue with the user used by the automatically triggered runs.

**Comment by stijnmoreels** [comment]
Close in favor of #252

**Pull request closed without merging**